tests/zfs-tests/tests/functional/no_space/enospc_df.ksh

```diff
@@ -0,0 +1,62 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2014, 2016 by Delphix. All rights reserved.
+# Copyright (c) 2019 by Datto Inc. All rights reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/tests/functional/no_space/enospc.cfg
+
+#
+# DESCRIPTION:
+# After filling a filesystem, the df command produces the
+# expected result against the pool root filesystem.
+#
+
+verify_runnable "both"
+
+function cleanup
+{
+	log_must_busy zpool destroy -f $TESTPOOL
```

Comment:
We should be using `default_cleanup_noexit` here, since `default_setup_noexit` was used below.

Reply:
I eliminated the cleanup function.. hope that's correct.  Log looks ok to me.
```
Test: /root/statfs/zfs/tests/zfs-tests/tests/functional/no_space/setup (run as root) [00:00] [PASS]
11:36:27.94 SUCCESS: truncate -s 100m /var/tmp/vdev_small
11:36:27.94 SUCCESS: truncate -s 512m /var/tmp/vdev_large
Test: /root/statfs/zfs/tests/zfs-tests/tests/functional/no_space/enospc_df (run as root) [00:15] [PASS]
11:36:27.98 ASSERTION: Correct df output is returned when file system is full.
11:36:27.98 NOTE: begin default_setup_noexit
11:36:28.18 SUCCESS: zpool create -f testpool /var/tmp/vdev_small
11:36:28.24 SUCCESS: zfs create testpool/testfs
11:36:28.31 SUCCESS: zfs set mountpoint=/var/tmp/testdir testpool/testfs
11:36:28.33 SUCCESS: zfs set compression=off testpool/testfs
11:36:28.33 NOTE: Writing file: testfile0 until ENOSPC.
11:36:43.18 write failed (-1), good_writes = 4996, error: No space left on device[28]
11:36:43.56 SUCCESS: zfs umount testpool/testfs
11:36:43.56 testpool        384K  128K  256K  34% /testpool
11:36:43.57 SUCCESS: eval df -h | grep testpool
11:36:43.62 df after ENOSPC works as expected.
11:36:43.62 NOTE: Performing local cleanup via log_onexit (default_cleanup_noexit)
11:36:43.83 SUCCESS: zpool destroy -f testpool
11:36:43.87 SUCCESS: rm -rf /var/tmp/testdir
Test: /root/statfs/zfs/tests/zfs-tests/tests/functional/no_space/cleanup (run as root) [00:00] [PASS]
11:36:44.06 SUCCESS: rm -f /var/tmp/vdev_small /var/tmp/vdev_large
```